lib/Listener/AddressCollectionListener.php

```diff
@@ -0,0 +1,68 @@
+<?php declare(strict_types=1);
+
+/**
+ * @copyright 2019 Christoph Wurst <christoph@winzerhof-wurst.at>
+ *
+ * @author 2019 Christoph Wurst <christoph@winzerhof-wurst.at>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+namespace OCA\Mail\Listener;
+
+use OCA\Mail\Events\MessageSentEvent;
+use OCA\Mail\Service\AutoCompletion\AddressCollector;
+use OCP\EventDispatcher\Event;
+use OCP\EventDispatcher\IEventListener;
+use OCP\ILogger;
+use Throwable;
+
+class AddressCollectionListener implements IEventListener {
+
+	/** @var AddressCollector */
+	private $collector;
+
+	/** @var ILogger */
+	private $logger;
+
+	public function __construct(AddressCollector $collector,
+								ILogger $logger) {
+		$this->collector = $collector;
+		$this->logger = $logger;
+	}
+
+	public function handle(Event $event): void {
+		if (!($event instanceof MessageSentEvent)) {
```

Comment:
How is that possible? https://github.com/nextcloud/mail/pull/1997/files#diff-de54363bb086805834528f6f9d0f58e7R118 the event is only listening for `MessageSentEvent`.

Reply:
While that is correct, neither my IDE nor Scrutinizer can statically know that and thus don't like any method invocations except for the base methods of `Event` ;)